src/extension.ts

```diff
@@ -204,41 +239,40 @@ export async function _runAndDisplayPathless(selections: vscode.Uri[], runInfo:
  * @param runInfo.outputChannel The output channel where information should be logged as needed
  * @param runInfo.commandName The specific command being run
  */
-export async function _runAndDisplayDfa(statusBarItem: vscode.StatusBarItem, runInfo: RunInfo): Promise<void> {
+export async function _runAndDisplayDfa(context:vscode.ExtensionContext ,runInfo: RunInfo, cancelToken: vscode.CancellationTokenSource): Promise<void> {
 	const {
 		outputChannel,
 		commandName
 	} = runInfo;
 	const startTime = Date.now();
 	try {
 		await verifyPluginInstallation();
-		// Set the Status Bar Item's text and un-hide it.
-		statusBarItem.text = messages.graphEngine.spinnerText;
-		statusBarItem.show();
-		// Get the targeted method.
-		const methodLevelTarget: string = await targeting.getSelectedMethod();
-		// Pull out the file from the target and use it to identify the project directory.
-		const currentFile: string = methodLevelTarget.substring(0, methodLevelTarget.lastIndexOf('#'));
-		const projectDir: string = targeting.getProjectDir(currentFile);
-		const results: string = await new ScanRunner().runDfa([methodLevelTarget], projectDir);
-		statusBarItem.hide();
-		if (results.length > 0) {
-			const panel = vscode.window.createWebviewPanel(
-				'dfaResults',
-				messages.graphEngine.resultsTab,
-				vscode.ViewColumn.Two,
-				{
-					enableScripts: true
-				}
-			);
-			panel.webview.html = results;
-		} else {
-			await vscode.window.showInformationMessage(messages.graphEngine.noViolationsFound);
+		if (await _shouldProceedWithDfaRun(context, outputChannel)) {
+			// Set the Status Bar Item's text and un-hide it.
+			// Get the targeted method.
+			const methodLevelTarget: string = await targeting.getSelectedMethod();
+			// Pull out the file from the target and use it to identify the project directory.
+			const currentFile: string = methodLevelTarget.substring(0, methodLevelTarget.lastIndexOf('#'));
+			const projectDir: string = targeting.getProjectDir(currentFile);
+			const results: string = await new ScanRunner().runDfa([methodLevelTarget], projectDir, context);
+			if (results.length > 0) {
+				const panel = vscode.window.createWebviewPanel(
+					'dfaResults',
+					messages.graphEngine.resultsTab,
+					vscode.ViewColumn.Two,
+					{
+						enableScripts: true
+					}
+				);
+				panel.webview.html = results;
+			} else {
+				cancelToken.cancel();
```

Comment:
This is a workaround to make the progress window to disappear. The way progress window works is that there is no cancel API and instead, you use the progress information to automatically cancel the window. Since we don't get such information from graph engine, the workaround is to create a cancel token and call that.

Reply:
Okay, but why did the `showInformationMessage` call get removed? Do we not need it anymore?

Reply:
It is not removed. It is now called as part of the cancel token invocation. See line number 99.